### PR TITLE
feat(search): add share action

### PR DESCRIPTION
**Pull request by cyndichin**
## Summary
Add share action that presents shared sheet for an item

## References 
IN-1122

## Implementation Details
* Created `ShareSheetView` using the `UIActivityController` since `ShareLink` is only available in iOS 16 and we can reuse our `ActionButton` easier
* Added Published property of `presentShareSheet` to trigger to show shareSheet
* Modified analytics to add scope details as a `componentDetail`

## Test Steps
- [ ] GIVEN search is active
AND a search has been performed
AND there are 1+ results
AND search results are visible
THEN every search result includes a share button

- [ ] GIVEN that a user taps on the share button
THEN the share sheet is presented as though the article had beens shared from Saved/Archived

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/PocketKit/MyList/ItemsList/ActivityViewController.swift [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import SwiftUI
+
+struct ShareSheetView: UIViewControllerRepresentable {
+    let activity: PocketActivity
+
+    func makeUIViewController(context: Context) -> UIActivityViewController {
+        let controller = UIActivityViewController(activity: activity)
+        controller.sheetPresentationController?.detents = [.medium()]
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/MyList/ItemsList/ActivityViewController.swift:8]: does not seem to make any difference in resolving the iOS 15 issue where the share sheet takes up the whole screen

File: PocketKit/Sources/PocketKit/MyList/ItemsList/SwiftUI/ListItem.swift [hunk 2]
```diff
@@ -27,7 +29,17 @@ struct ListItem: View {
...
                 ActionButton(viewModel.favoriteAction(index: viewModel.index, scope: scope), selected: viewModel.isFavorite)
-                ActionButton(viewModel.shareAction)
+
+                ActionButton(viewModel.shareAction(index: viewModel.index, scope: scope))
+                    .sheet(isPresented: $viewModel.presentShareSheet) {
+                        if #available(iOS 16.0, *) {
+                            ShareSheetView(activity: PocketItemActivity(url: viewModel.item.url))
+                                .presentationDetents([.medium])
+                        } else {
+                            ShareSheetView(activity: PocketItemActivity(url: viewModel.item.url))
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/MyList/ItemsList/SwiftUI/ListItem.swift:39]: issue where the share sheet takes up the whole screen for iOS 15

File: PocketKit/Sources/Analytics/AppEvents/Search.swift [hunk 3]
```diff
@@ -45,21 +47,33 @@ public extension Events.Search {
...
-            identifier = "global-nav.search.archive.unfavorite"
-        case .all:
-            identifier = "global-nav.search.all.unfavorite"
-        }
+        return Engagement(
+            .general,
+            uiEntity: UiEntity(
+                .button,
+                identifier: "global-nav.search.unfavorite",
+                componentDetail: getScopeIdentifier(scope: scope),
```
Comment (cyndichin) [inline PocketKit/Sources/Analytics/AppEvents/Search.swift:55]: conversation on analytics ongoing here: https://pocket.slack.com/archives/C01UH57EJKD/p1676996499616879

will refine analytics in a different PR

File: PocketKit/Sources/PocketKit/MyList/ItemsList/SwiftUI/ListItem.swift (hunk 2 above)
Comment (cyndichin) [reply PocketKit/Sources/PocketKit/MyList/ItemsList/SwiftUI/ListItem.swift:39]: adding comment as weird bug, convo here: https://pocket.slack.com/archives/C01UH57EJKD/p1677166224894789

File: PocketKit/Sources/PocketKit/MyList/ItemsList/PocketItemViewModel.swift [hunk 4]
```diff
@@ -66,13 +69,22 @@ class PocketItemViewModel: ObservableObject {
             return
         }
 
-        tracker.track(event: Events.Search.favoriteItem(itemUrl: savedItem.url, positionInList: index, scope: scope))
+        tracker.track(event: Events.Search.unfavoriteItem(itemUrl: savedItem.url, positionInList: index, scope: scope))
```
Comment (cyndichin) [inline PocketKit/Sources/PocketKit/MyList/ItemsList/PocketItemViewModel.swift:72]: woops 😄

**Review by bassrock — APPROVED** [review]
ship it!

**Review by Gio2018 — APPROVED** [review]
Tested on iPhone, iPad and a few simulators and worked for me

**Pull request merged**